pkg/ingester/ingester_ring.go

```diff
@@ -18,7 +19,10 @@ import (
 const (
 	// sharedOptionWithRingClient is a message appended to all config options that should be also
 	// set on the components running the ingester ring client.
-	sharedOptionWithRingClient = " This option needs be set on ingesters, distributors, queriers and rulers when running in microservices mode."
+	sharedOptionWithRingClient      = " This option needs be set on ingesters, distributors, queriers and rulers when running in microservices mode."
+	tokenGenerationStrategyFlag     = "token-generation-strategy"
+	randomTokenGeneration           = "random-tokens"
+	spreadMinimizingTokenGeneration = "spread-min-tokens"
```

Comment:
[nit] A couple of suggestions:
1. No need for the `-tokens` suffix (the config param is called `token-generation-strategy` so looks clear
2. I suggest to be a bit more verbose about "min" for clarity

```suggestion
	randomTokenGeneration           = "random"
	spreadMinimizingTokenGeneration = "spread-minimizing"
```